File: app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/acoustid/Release.java

package org.metabrainz.mobile.data.sources.api.entities.acoustid;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class Release {

    @SerializedName("country")
    @Expose
    private String country;
    @SerializedName("date")
    @Expose
    private Date date;
    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("medium_count")
    @Expose
    private int mediumCount;
    @SerializedName("mediums")
    @Expose
    private List<Medium> mediums = new ArrayList<>();
    @SerializedName("releaseEvents")
    @Expose
    private List<ReleaseEvent> releaseEvents = new ArrayList<>();
    @SerializedName("title")
    @Expose
    private String title;
    @SerializedName("track_count")
    @Expose
    private int trackCount;

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public int getMediumCount() {
        return mediumCount;
    }

    public void setMediumCount(int mediumCount) {
        this.mediumCount = mediumCount;
    }

    public List<Medium> getMediums() {
        return mediums;
    }

    public void setMediums(List<Medium> mediums) {
        this.mediums = mediums;
    }

    public List<ReleaseEvent> getReleaseEvents() {
        return releaseEvents;
    }

    public void setReleaseEvents(List<ReleaseEvent> releaseEvents) {
        this.releaseEvents = releaseEvents;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getTrackCount() {
        return trackCount;
    }

    public void setTrackCount(int trackCount) {
        this.trackCount = trackCount;
    }

}

File: app/src/main/java/org/metabrainz/android/model/acoustid/Release.kt
package org.metabrainz.android.model.acoustid

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

class Release {
    @SerializedName("country")
    @Expose
    var country: String? = null

    @SerializedName("date")
    @Expose
    var date: org.metabrainz.android.model.acoustid.Date? = null

    @SerializedName("id")
    @Expose
    var id: String? = null

    @SerializedName("medium_count")
    @Expose
    var mediumCount = 0

    @SerializedName("mediums")
    @Expose
    var mediums: List<org.metabrainz.android.model.acoustid.Medium> = ArrayList()

    @SerializedName("releaseEvents")
    @Expose
    var releaseEvents: List<org.metabrainz.android.model.acoustid.ReleaseEvent> = ArrayList()

    @SerializedName("title")
    @Expose
    var title: String? = null

    @SerializedName("track_count")
    @Expose
    var trackCount = 0
}